src/Features/CSharp/Portable/BraceMatching/BlockCommentBraceMatcher.cs

```diff
@@ -14,33 +14,52 @@
 namespace Microsoft.CodeAnalysis.CSharp.BraceMatching;
 
 [ExportBraceMatcher(LanguageNames.CSharp), Shared]
-internal class BlockCommentBraceMatcher : IBraceMatcher
+[method: ImportingConstructor]
+[method: Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+internal sealed class BlockCommentBraceMatcher() : IBraceMatcher
 {
-    [ImportingConstructor]
-    [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
-    public BlockCommentBraceMatcher()
-    {
-    }
-
     public async Task<BraceMatchingResult?> FindBracesAsync(Document document, int position, BraceMatchingOptions options, CancellationToken cancellationToken)
     {
         var root = await document.GetRequiredSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+        var token = root.FindToken(position, findInsideTrivia: false);
+        if (token == default)
+            return null;
+
+        var span = token.Span;
+        if (span.Contains(position))
+            return null;
+
         var text = await document.GetValueTextAsync(cancellationToken).ConfigureAwait(false);
```

Comment:
>var text = await document.GetValueTextAsync(cancellationToken).ConfigureAwait(false); [](http://example.com/codeflow?start=8&length=85)

seems like a bit of a shame to do this when it's almost never used, right?

Reply:
yeah.  but it should be near free.  for open docs, it literally is getting the ITextSnapshot wrapper we are sure to have.